### ISSUE TITLE
release 2.4.24 broken - UnknownConditionError

### ISSUE THREAD

**Issue by fuzziness** [binarylogic/searchlogic#113]
The condition 'description_like_before_type_cast' is not a valid condition, we could not find any scopes that match this.
Extracted source (around line #16):

<code><pre>
13:         &lt;ul&gt;
14:           &lt;li&gt;
15:             &lt;span&gt;
16:               &lt;%= form.input :name_or_description_like, :label =&gt; 'articolo',
17:                 :label_html =&gt; {:class =&gt; 'desc'},
18:                 :hint =&gt; 'codice o descrizione',
19:                 :wrapper =&gt; false, :required =&gt; false %&gt;
</pre></code>

**Comment by ghost** [comment]
Are you sure you're using 2.4.24? I believe the latest release fixes this.

**Comment by fuzziness** [comment]
yepp... just tried to rolling back to 2.4.23 an all works fine (rails 2.3.8 - ruby 1.8.7)

**Comment by frahugo** [comment]
I have the same issue with 2.3.8 and 1.8.7 and reverting to 2.4.23 solved it.

**Comment by adamcrown** [comment]
I'm having the same problem. Rolling back to 2.4.23 does solve the problem.

The issue seems to be that `Search.respond_to?()` was added in 2.4.24 (SHA: 21770e980ea2f2e3304f1a5b99e81a0f382d5bc0) but the `respond_to?()` calls `interpolate_or_conditions()` which raises an `UnknownConditionError` rather than returning false when it's passed an invalid "or condition".

I was able to reproduce it in the console with
    User.search().or_condition?('foo_or_bar_like')
